### PR TITLE
update getting-started guide

### PR DESCRIPTION
**Pull request by rksharma95**
ref: #946 
Signed-off-by: rksharma95 <ramakant@accuknox.com>

### CODE REVIEW

File: getting-started/deployment_guide.md [hunk 1]
```diff
@@ -36,42 +38,179 @@ KubeArmor Policy Manager Deployment ...
 KubeArmor Host Policy Manager Service ...
 KubeArmor Host Policy Manager Deployment ...
 ```
+
 </details>
 
-It is assumed that the k8s cluster is already present/reachable setup with the [*required prerequisites*](#Prerequisites) and the user has rights to create service-accounts and cluster-role-bindings.
+It is assumed that the k8s cluster is already present/reachable setup with the [_required prerequisites_](#Prerequisites) and the user has rights to create service-accounts and cluster-role-bindings.
+
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]:
```suggestion
```

File: getting-started/deployment_guide.md [hunk 2]
```diff
@@ -9,11 +9,13 @@ Check the [KubeArmor support matrix](support_matrix.md) to verify if your platfo
 ## Deployment Steps
 
 ### 1. Download and install karmor cli-tool
+
 ```
 curl -sfL http://get.kubearmor.io/ | sudo sh -s -- -b /usr/local/bin
 ```
 
 ### 2. Install KubeArmor
+
```
Comment (nyrahul) [inline getting-started/deployment_guide.md:18]:
```suggestion
```

File: getting-started/deployment_guide.md [hunk 3]
```diff
@@ -9,11 +9,13 @@ Check the [KubeArmor support matrix](support_matrix.md) to verify if your platfo
 ## Deployment Steps
 
 ### 1. Download and install karmor cli-tool
+
```
Comment (nyrahul) [inline getting-started/deployment_guide.md:12]:
```suggestion
```

File: getting-started/deployment_guide.md [hunk 4]
```diff
@@ -36,42 +38,179 @@ KubeArmor Policy Manager Deployment ...
...
-### 3. Deploying sample app and policies
-   
-#### a. Deploy sample [multiubuntu app](../examples/multiubuntu.md)
 ```
-kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/multiubuntu/multiubuntu-deployment.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/wordpress-mysql/wordpress-mysql-deployment.yaml
 ```
 
-#### b. Deploy [sample policies](security_policy_examples.md)
+### 4. Use case and deploying Security Policy
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: Please use collapsible markdown sections to create usecases and karmor log details below.

File: getting-started/deployment_guide.md [hunk 5]
```diff
@@ -36,42 +38,179 @@ KubeArmor Policy Manager Deployment ...
...
+
+#### **a. Use Case**
```
Comment (nyrahul) [inline getting-started/deployment_guide.md]: Use case should have an appropriate title explaining use-case.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: getting-started/deployment_guide.md [hunk 6]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+  selector:
+    matchLabels:
+      app: mysql
+  file:
+    matchDirectories:
+      - dir: /var/lib/mysql/
+        recursive: true
+  action: Audit
+```
+
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:110]: open a new terminal session as the current one is now executing the karmor log process

File: getting-started/deployment_guide.md [hunk 7]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/wordpress-mysql/security-policies/ksp-mysql-audit-dir.yaml
+```
+
+#### **Start Observing Alerts**
+
+Before simulating the policy violation in the next step execute the following command in a new terminal/tab to start observing KubeArmor logs/alerts, output for the same will be discussed in upcoming section `Getting Alerts/Telemetry from KubeArmor`.
+
+```
+$ karmor logs
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md]: $ karmor log

File: getting-started/deployment_guide.md [hunk 8]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+#### **Simulate policy violation**
+
+```
+$ POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=mysql" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@mysql-6c6fcdccf-v6qhz:/# touch /var/lib/mysql/test-1
+root@mysql-6c6fcdccf-v6qhz:/# ls /var/lib/mysql/
+auto.cnf     ib_logfile1  mysql		      test    wordpress
+ib_logfile0  ibdata1	  performance_schema  test-1
+
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:132]: root@mysql-6c6fcdccf-v6qhz:/#  exit

File: getting-started/deployment_guide.md [hunk 9]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+There will'be no alert generated by the KubeArmor for any access to the `/var/lib/mysql` directory or subdirectories.
+
+```
+$ POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=mysql" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@mysql-6c6fcdccf-v6qhz:/# touch /var/lib/mysql/test
+root@mysql-6c6fcdccf-v6qhz:/# ls /var/lib/mysql/
+auto.cnf     ib_logfile1  mysql		      test    wordpress
+ib_logfile0  ibdata1	  performance_schema
+```
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md]: root@mysql-6c6fcdccf-v6qhz:/# exit
```

File: getting-started/deployment_guide.md [hunk 10]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+
+A pod may contain unprotected files with sensitive data that need to be protected from unautherized access. In our sample wordpress app, wordpress pod contains a file `wp-config.php` that has sensitive auth credentials. KubeArmor security policy can be applied to block access to this file from unknown processes.
+
+**Before Applying any Policy**
+
+```
+POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=wordpress" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@wordpress-84dbf54bb8-8kczs:/var/www/html# cat /var/www/html/wp-config.php
+
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:193]: root@wordpress-84dbf54bb8-8kczs:/var/www/html# exit

File: getting-started/deployment_guide.md [hunk 11]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/wordpress-mysql/security-policies/ksp-wordpress-block-config.yaml
+```
+
+#### **Start Observing Alerts**
+
+Before simulating the policy violation in the next step execute the following command in a new terminal/tab to start observing KubeArmor logs/alerts, output for the same will be discussed in upcoming section `Getting Alerts/Telemetry from KubeArmor`.
+
+```
+$ karmor logs
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md]: $ karmor log

File: getting-started/deployment_guide.md [hunk 12]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+```
+
+#### **Simulate policy violation**
+
+```
+$ POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=wordpress" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@wordpress-84dbf54bb8-8kczs:/var/www/html# cat /var/www/html/wp-config.php
+cat: /var/www/html/wp-config.php: Permission denied
+```
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:344]: root@wordpress-84dbf54bb8-8kczs:/var/www/html# exit
```

File: getting-started/deployment_guide.md [hunk 13]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
-#### c. Simulate policy violation
+#### **Start Observing Alerts**
+
+Before simulating the policy violation in the next step execute the following command in a new terminal/tab to start observing KubeArmor logs/alerts, output for the same will be discussed in upcoming section `Getting Alerts/Telemetry from KubeArmor`.
+
 ```
-$ POD_NAME=$(kubectl get pods -n multiubuntu -l "group=group-1,container=ubuntu-1" -o jsonpath='{.items[0].metadata.name}') && kubectl -n multiubuntu exec -it $POD_NAME -- bash
-# sleep 1
-(Permission Denied)
+$ karmor logs
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md]: $ karmor log

File: getting-started/deployment_guide.md [hunk 14]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
-#### a. Enable port-forwarding for KubeArmor relay (if needed)
+#### **Simulate policy violation**
+
 ```
-kubectl port-forward -n kube-system svc/kubearmor 32767:32767
+$ POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=wordpress" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@wordpress-84dbf54bb8-8kczs:/var/www/html# cat /run/secrets/kubernetes.io/serviceaccount/token
+cat: /run/secrets/kubernetes.io/serviceaccount/token: Permission denied
 ```
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:453]: root@wordpress-84dbf54bb8-8kczs:/var/www/html# exit
```

File: getting-started/deployment_guide.md [hunk 15]
```diff
@@ -36,42 +38,618 @@ KubeArmor Policy Manager Deployment ...
...
+#### **Simulate policy violation**
+
+```
+$ POD_NAME=$(kubectl get pods -n wordpress-mysql -l "app=wordpress" -o jsonpath='{.items[0].metadata.name}') && kubectl -n wordpress-mysql exec -it $POD_NAME -- bash
+
+root@wordpress-84dbf54bb8-8kczs:/var/www/html# apt
+bash: /usr/bin/apt: Permission denied
+root@wordpress-84dbf54bb8-8kczs:/var/www/html# apt-get update
+bash: /usr/bin/apt-get: Permission denied
+
```
Comment (Pietrokechi) [inline getting-started/deployment_guide.md:596]: root@wordpress-84dbf54bb8-8kczs:/var/www/html# exit

**Review by Pietrokechi — CHANGES_REQUESTED** [review]
@rksharma95 gone through the process and done my review with some suggested changes thanks

File: getting-started/deployment_guide.md (hunk 7 above)
Comment (rksharma95) [reply getting-started/deployment_guide.md]: thanks for pointing out this typo

File: getting-started/deployment_guide.md (hunk 6 above)
Comment (rksharma95) [reply getting-started/deployment_guide.md:110]: we are mentioning it in `Start Observing Alerts` section

**Comment by nyrahul** [comment]
Lets remove the section "Manual YAML based KubeArmor deployment" from the Getting started guide. Please fix the spelling mistakes in the newly added use-cases section. Can you use grammarly or similar tool to check for grammatical mistakes? @rksharma95

**Comment by rksharma95** [comment]
handled with #1263

**Pull request closed without merging**